hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -261,6 +287,50 @@ public List<WriteStatus> compact(HoodieCompactionHandler compactionHandler,
     }).collect(toList());
   }
 
+  /**
+   * Execute a single compaction operation and report back status.
+   */
+  public List<WriteStatus> compact(HoodieCompactionHandler compactionHandler,
+                                   HoodieTableMetaClient metaClient,
+                                   HoodieWriteConfig config,
+                                   CompactionOperation operation,
+                                   String instantTime,
+                                   String maxInstantTime,
+                                   Option<InstantRange> instantRange,
+                                   TaskContextSupplier taskContextSupplier,
+                                   CompactionExecutionHelper executionHelper,
+                                   Option<EngineBroadcastManager> broadcastManagerOpt) throws IOException {
+    List<WriteStatus> writeStatusList = compactionHandler.runCompactionUsingFileGroupReader(instantTime,
+        operation, broadcastManagerOpt.get().retrieveFileGroupReaderContext(metaClient.getBasePath()).get(),
+        broadcastManagerOpt.get().retrieveStorageConfig().get());
+    writeStatusList
+        .forEach(s -> {
+          final HoodieWriteStat stat = s.getStat();
+          /*
+          fill in log reading stats
+          stat.setTotalUpdatedRecordsCompacted(scanner.getNumMergedRecordsInLog());
```

Comment:
why commented out?
if not required anymore, we can remove it

Reply:
Removed and the log reader stats functionality is now in the file group reader.